### PR TITLE
Port to spago (CI)

### PR DESCRIPTION
**Pull request by emiel**
Here is my attempt to continue with the work done here https://github.com/exercism/purescript/pull/127 in replacing bower/pulp with spago. I focused on getting CI setup to build and test the exercises (just practice exercises for now). The approach is more or less similar to what was being done for bower. Spago uses two files `packages.dhall` (defines package-set) and `spago.dhall` (project dependencies).

* All exercises should use the same package set so that we can share a .spago and output directory.
* We gain some additional flexibility in that spago.dhall can vary per exercise. This is nice because we can whittle down the exercise dependencies to the bare minimum and we save students some diskspace. Also spago has started warning about unused dependencies... a topic for when we tackle porting the exercises to 0.14.x.

Please have a look (the last commit contains the interesting bits) and see if you think this approach is worth pursuing. If so I can continue to update the docs and add the finishing touches.

NOTE: Without sharing the `output` and `.spago` directories the CI run takes around 7 minutes...

@connorads @aimorris

### CODE REVIEW

**Comment by emiel** [comment]
Here is an example of a CI run: https://github.com/emiel/purescript/runs/2442322095?check_suite_focus=true

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        version: [0.13.8]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Use PureScript compiler ${{ matrix.version }}
+        uses: purescript-contrib/setup-purescript@main
+        with:
+          purescript: ${{ matrix.version }}
+          spago: "0.19.2"
```
Comment (emiel) [inline .github/workflows/ci.yml:48]: I pin spago to "0.19.2" because later versions don't appear to run the tests. :(

**Comment by iHiD** [comment]
@emiel Thank you 💙 
@ErikSchierboom will take a look at this later in the week when he's back working. Before then hopefully some @exercism/purescript people can take a look! :)

**Review by SleeplessByte — APPROVED** [review]

**Comment by SleeplessByte** [comment]
The travis is failing because it's still using the old configlet

**Comment by emiel** [comment]
> The travis is failing because it's still using the old configlet

Thanks for having a look @SleeplessByte. It is my understanding Excercism is migrating completely to GitHub CI and thus that is where my focus is now. I'd rather tackle configlet in another PR because I guess there is more work to do there. ;)

**Comment by SleeplessByte** [comment]
Totally. It was mostly informational and why I approved.

**Review by ErikSchierboom — APPROVED** [review]
I don't know anytthing about purescript or dhall, but it looks good to me!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ErikSchierboom) [reply .github/workflows/ci.yml:48]: This will be good to include in the documentation :)

**Comment by aimorris** [comment]
I concur

File: .github/workflows/ci.yml (hunk 1 above)
Comment (emiel) [reply .github/workflows/ci.yml:48]: This has been resolved in 0.20.2 and is no longer an issue.

**Comment by emiel** [comment]
Closing in favor of https://github.com/exercism/purescript/pull/185

**Pull request closed without merging**